src/jaxsim/api/contact.py

```diff
@@ -137,9 +137,17 @@ def collidable_point_dynamics(
     # all collidable points belonging to the robot.
     W_p_Ci, W_ṗ_Ci = js.contact.collidable_point_kinematics(model=model, data=data)
 
+    # Import privately the soft contacts classes.
+    from jaxsim.rbda.contacts.soft import SoftContacts, SoftContactsState
+
     # Build the soft contact model.
     match model.contact_model:
-        case s if isinstance(s, SoftContacts):
+
+        case SoftContacts():
+
+            assert isinstance(model.contact_model, SoftContacts)
```

Comment:
This is to help resources for type hinting like mypy and IDEs. Otherwise they cannot assume the type of `model.contact_model`.

Reply:
Okay so I guess this should be part of #190, right? Anyhow we can merge it like this

Reply:
I don't have plans to work on mypy anytime soon, the issue is mainly a tracker if anyone wants to give it a shot. Curiosity, does your VSCode autocomplete `model.contact_model.parameters`<kbd>Tab</kbd>`K|D|mu`? PyCharm is able to autocomplete (and remove the warnings) only with the asserts.

This is a consequence of having `model.contact_model` typed with the base class `jaxsim.rbda.ContactModel`.

Reply:
VSCode doesn't autocomplete with K,D,mu regardless your change, but iPython does, with or without the assert

Reply:
Ok good to know about VSCode, I always had the impression that PyCharm uses a more advanced engine. IPython is somewhat different because it has access to the runtime variables that can be easily introspected. The IDEs can only access to the textual information, i.e. no runtime information is used. This is because in some occasion you see in Python code either asserts like the one discussed here ([type narrowing](https://mypy.readthedocs.io/en/stable/type_narrowing.html)) or explicit statements like `var: VarType` (that cannot be applied to non-self attributes like in this case).